### PR TITLE
Fix bracket-if matchit patterns

### PR DESCRIPTION
**Pull request by dkearns**
- Wrap bracket-if matchit patterns in word boundaries.
- Add some ftplugin tests.

### CODE REVIEW

**Comment by jkotlinski** [comment]
Sorry, I am busy with real-life things. I can have a look in a week or two,
if that is not too long for you.

ons 7 juni 2023 kl. 13:18 skrev Doug Kearns ***@***.***>:

**Comment by dkearns** [comment]
No hurry.  It's not going anywhere.

File: test.fs [hunk 1]
```diff
@@ -362,5 +362,69 @@ XC!+ XC!+? XC@+
...
+\ TEST:    Matchit functionality for conditional compilation
+\   words.
+\ EXECUTE: With the cursor intially positioned on the "[IF]" in the
+\   example, execute the Vim % command repeatedly.
+\ RESULT:  The cursor should cycle between "[IF]", "[ELSE]" and
+\   "[THEN]" ignoring those words contained in strings and
+\   comments.
+
+0 [IF]
+    S" skip these [ELSE] [THEN]"
```
Comment (jkotlinski) [inline test.fs:397]: This test seems to fail for me. Do you get the same?

File: test.fs [hunk 2]
```diff
@@ -362,5 +362,69 @@ XC!+ XC!+? XC@+
...
+42 CONSTANT MEANING
+MEANING \ <- execute [d here
+
+\ TEST:    Test definition searches in included files.
+\ EXECUTE: With the cursor on the word ELSEWHERE1 execute the
+\   Vim [<C-D> command.
+\ RESULT:  The definition of ELSEWHERE1 should be displayed in
+\   the message line.
+
+INCLUDE input/external1.fs ELSEWHERE1 \ <- execute [<C-D> here
```
Comment (jkotlinski) [inline test.fs:419]: Is the description of the test wrong? [<C-D> jumps to external1.fs . [d will display on the message line.

File: test.fs [hunk 3]
```diff
@@ -362,5 +362,69 @@ XC!+ XC!+? XC@+
...
+
+\ TEST:    Test definition searches in included files without a
+\   filename extension.
+\ EXECUTE: With the cursor on the word ELSEWHERE execute the Vim
+\   [<C-D> command.
+\ RESULT:  The definition of ELSEWHERE2 should be displayed in
+\   the message line.
+
+INCLUDE input/external2 ELSEWHERE2 \ <- execute [<C-D> here
```
Comment (jkotlinski) [inline test.fs:428]: Also wondering if the description is wrong. 
`[<C-D>` jumps to external1.fs . [d will display on the message line.

**Comment by jkotlinski** [comment]
Sorry, it was a bit premature to merge before running the tests..!

File: test.fs (hunk 1 above)
Comment (dkearns) [reply test.fs:397]: Yes, this one probably depends on #25.  If you change it to `1 [IF]` it should work.

File: test.fs (hunk 2 above)
Comment (dkearns) [reply test.fs:419]: Yep

File: test.fs (hunk 3 above)
Comment (dkearns) [reply test.fs:428]: And yep again.   I'll fix them.

**Comment by dkearns** [comment]
No problem.  If I didn't leave it as a draft it's my fault.

I think I was mostly interested in seeing if that's what you had in mind for the manual testing.

**Comment by jkotlinski** [comment]
Yes, that is what I had in mind. I suppose it is not the ideal way to do testing, and an automated setup is better. But, for me it is already good.

**Pull request merged**